### PR TITLE
snmptrap_snmpuser_binding: add check for resource

### PR DESCRIPTION
**Pull request by ChrisM3rsi**
Signed-off-by: christosm <christos.mersinis@citrix.com>

### CODE REVIEW

File: service/config.go [hunk 1]
```diff
@@ -336,8 +336,15 @@ func (c *NitroClient) DeleteResource(resourceType string, resourceName string) e
 //DeleteResourceWithArgs deletes a resource of supplied type and name. Args are supplied as an array of strings
 //Each array entry is formatted as "key:value"
 func (c *NitroClient) DeleteResourceWithArgs(resourceType string, resourceName string, args []string) error {
+	//Remove unwanted argument (username) for listing but keep it for delete operation
+	var err error
+	if resourceType == "snmptrap_snmpuser_binding" {
+		get_args := args[:len(args)-1]
```
Comment (sumanth-lingappa) [inline service/config.go:342]: Please iterate over the `args` array and remove the item with the `username` key.

**Review by sumanth-lingappa — CHANGES_REQUESTED** [review]

File: service/config.go [hunk 2]
```diff
@@ -336,8 +336,20 @@ func (c *NitroClient) DeleteResource(resourceType string, resourceName string) e
 //DeleteResourceWithArgs deletes a resource of supplied type and name. Args are supplied as an array of strings
 //Each array entry is formatted as "key:value"
 func (c *NitroClient) DeleteResourceWithArgs(resourceType string, resourceName string, args []string) error {
-
-	_, err := c.listResourceWithArgs(resourceType, resourceName, args)
+	//Remove unwanted argument (username) for listing but keep it for delete operation
```
Comment (sumanth-lingappa) [inline service/config.go:339]: this comment should come inside the `if` block

File: service/config.go [hunk 3]
```diff
@@ -336,8 +336,20 @@ func (c *NitroClient) DeleteResource(resourceType string, resourceName string) e
...
+	var err error
+	if resourceType == "snmptrap_snmpuser_binding" {
+		get_args := make([]string, 0)
```
Comment (sumanth-lingappa) [inline service/config.go:342]: rename `get_args` to a meaningful variable name. eg., `argsWithoutUsername`

File: service/config.go [hunk 4]
```diff
@@ -336,8 +336,20 @@ func (c *NitroClient) DeleteResource(resourceType string, resourceName string) e
...
+		for i, v := range args {
+			if !strings.Contains(v, "username:") {
+				//append all elements except username
```
Comment (sumanth-lingappa) [inline service/config.go:345]: This comment is not required. Your code explains it.
Maybe you can explain the intention behind it. Not what your code is doing.

**Review by sumanth-lingappa — CHANGES_REQUESTED** [review]

**Review by sumanth-lingappa — APPROVED** [review]

**Comment by imkritesh** [comment]
Hello @sumanth-lingappa @ChrisM3rsi ,

I wanted to get context on why do we need to remove username parameter for snmptrap_snmpuser_binding  while doing get request ?

**Pull request merged**